File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/implementation/mappers/HospitalServiceMapper.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.implementation.mappers;

import io.zenwave360.example.clinicaltool.common.mappers.BaseMapper;
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*;
import java.util.List;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;
import org.mapstruct.factory.Mappers;

@Mapper(uses = {BaseMapper.class})
public interface HospitalServiceMapper {

    HospitalServiceMapper INSTANCE = Mappers.getMapper(HospitalServiceMapper.class);

    // input mappings
    // Doctor-Doctor updateDoctor
    Doctor asDoctor(Doctor input);

    @Mapping(target = "id", ignore = true)
    Doctor update(@MappingTarget Doctor entity, Doctor input);
    // HospitalIdnull-Doctor listHospitalDoctors
    Doctor asDoctor(Long hospitalId);

    @Mapping(target = "id", ignore = true)
    Doctor update(@MappingTarget Doctor entity, Long hospitalId);
    // Hospital-Hospital updateHospital
    Hospital asHospital(Hospital input);

    @Mapping(target = "id", ignore = true)
    Hospital update(@MappingTarget Hospital entity, Hospital input);
    // HospitalIdnull-Hospital listHospitalPatients
    Hospital asHospital(Long hospitalId);

    @Mapping(target = "id", ignore = true)
    Hospital update(@MappingTarget Hospital entity, Long hospitalId);
    // output mappings
    // Hospital-PatientHospital listHospitalPatients
    PatientHospital asPatientHospital(Hospital entity);

    List<PatientHospital> asPatientHospitalList(List<Hospital> entity);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/implementation/mappers/HospitalServiceMapper.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.implementation.mappers

import io.zenwave360.example.clinicaltool.common.mappers.BaseMapper
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingTarget
import org.mapstruct.factory.Mappers

@Mapper(uses = [BaseMapper::class])
interface HospitalServiceMapper {

    companion object {
        val INSTANCE: HospitalServiceMapper = Mappers.getMapper(HospitalServiceMapper::class.java)
    }

    // input mappings
    // Doctor-Doctor updateDoctor

    fun asDoctor(input: Doctor): Doctor

    @Mapping(target = "id", ignore = true) fun update(@MappingTarget entity: Doctor, input: Doctor): Doctor

    // HospitalIdnull-Doctor listHospitalDoctors
    fun asDoctor(hospitalId: Long): Doctor {
        return Doctor().apply {
            // TODO: implement this method
            // this.hospitalId = hospitalId
        }
    }

    fun update(entity: Doctor, hospitalId: Long): Doctor {
        return entity.apply {
            // TODO: implement this method
            // this.hospitalId = hospitalId
        }
    }

    // Hospital-Hospital updateHospital

    fun asHospital(input: Hospital): Hospital

    @Mapping(target = "id", ignore = true) fun update(@MappingTarget entity: Hospital, input: Hospital): Hospital

    // HospitalIdnull-Hospital listHospitalPatients
    fun asHospital(hospitalId: Long): Hospital {
        return Hospital().apply {
            // TODO: implement this method
            // this.hospitalId = hospitalId
        }
    }

    fun update(entity: Hospital, hospitalId: Long): Hospital {
        return entity.apply {
            // TODO: implement this method
            // this.hospitalId = hospitalId
        }
    }

    // output mappings
    // Hospital-PatientHospital listHospitalPatients
    fun asPatientHospital(entity: Hospital): PatientHospital

    fun asPatientHospitalList(entity: List<Hospital>): List<PatientHospital>
}
